fix macos drop privs

phillyqueso opened the pull request:
fixes #205 

references:
- https://blog.lxsang.me/post/id/28 <-- nudged to use [nix](https://docs.rs/nix/0.18.0/nix/unistd/index.html).
- https://github.com/capistrano/capistrano/issues/1520 <-- helped understand the underlying macos specific issue.

Noah-Kennedy commented:
@phillyqueso run `cargo fmt`

Noah-Kennedy commented:
@phillyqueso so I have good news and bad news. The good news is that we fixed the CI issues on the master branch, and all you need to do here is merge in those changes. The bad news is that there are some mean merge conflicts here, as we just split the crate into a workspace, which should have been done long ago.

phillyqueso commented:
> @phillyqueso so I have good news and bad news. The good news is that we fixed the CI issues on the master branch, and all you need to do here is merge in those changes. The bad news is that there are some mean merge conflicts here, as we just split the crate into a workspace, which should have been done long ago.

No problem @Noah-Kennedy! I'll rebase and push the fix soon.

Noah-Kennedy reviewed (approved):
Thanks for the PR!

jiegec commented on boringtun/src/device/drop_privileges.rs:
```diff
@@ -5,23 +5,35 @@ use crate::device::errno_str;
-        return Err(Error::DropPrivileges("NULL from getlogin".to_owned()));
+    #[cfg(target_os = "macos")]
+    {
+        let uname: &'static str = env!("USER");
```
env!() retrieves the environment variable at compile time, is it expected? If I compile the code in one user, and use the binary in another user(or machine), the behavior is incorrect.

Noah-Kennedy replied:
I'll open up a PR after I ear breakfast.

Noah-Kennedy replied:
Also, next time, it's much easier to track this if you create an issue to track the problem.

Noah-Kennedy replied:
Fixed
